cli/src/Api.ts

```diff
@@ -232,24 +272,174 @@ export default class Api {
+            roleAccout: workerApplication.role_account,
+            stakes: {
+                application: appStakingId.isSome ? (await this.stakeValue(appStakingId.unwrap())).toNumber() : 0,
+                role: roleStakingId.isSome ? (await this.stakeValue(roleStakingId.unwrap())).toNumber() : 0
+            },
+            humanReadableText: application.human_readable_text.toString(),
+            stage: application.stage.type as ApplicationStageKeys
+        };
+    }
+
+    async groupApplication(group: WorkingGroups, workerApplicationId: number): Promise<GroupApplication> {
+        const workerApplication = await this.workerApplicationById(group, workerApplicationId);
+        return await this.parseApplication(workerApplicationId, workerApplication);
+    }
+
+    protected async groupOpeningApplications(group: WorkingGroups, workerOpeningId: number): Promise<GroupApplication[]> {
+        const applications: GroupApplication[] = [];
+
+        const nextAppId = await this.workingGroupApiQuery(group).nextWorkerApplicationId() as WorkerApplicationId;
+        for (let i = 0; i < nextAppId.toNumber(); i++) {
+            const workerApplication = await this.workerApplicationById(group, i);
+            if (workerApplication.worker_opening_id.toNumber() !== workerOpeningId) {
+                continue;
+            }
+            applications.push(await this.parseApplication(i, workerApplication));
+        }
+
+
+        return applications;
+    }
+
+    async groupOpening(group: WorkingGroups, workerOpeningId: number): Promise<GroupOpening> {
+        const nextId = ((await this.workingGroupApiQuery(group).nextWorkerOpeningId()) as WorkerOpeningId).toNumber();
+
+        if (workerOpeningId < 0 || workerOpeningId >= nextId) {
+            throw new CLIError('Invalid group opening ID!');
+        }
+
+        const groupOpening = this.singleLinkageResult<WorkerOpening>(
+            await this.workingGroupApiQuery(group).workerOpeningById(workerOpeningId) as LinkageResult
+        );
+
+        const openingId = groupOpening.opening_id.toNumber();
+        const opening = await this.hiringOpeningById(openingId);
+        const applications = await this.groupOpeningApplications(group, workerOpeningId);
+        const stage = await this.parseOpeningStage(opening.stage);
+        const stakes = {
+            application: opening.application_staking_policy.unwrapOr(undefined),
+            role: opening.role_staking_policy.unwrapOr(undefined)
+        }
+
+        return ({
+            workerOpeningId,
+            openingId,
+            opening,
+            stage,
+            stakes,
+            applications
+        });
+    }
+
+    async parseOpeningStage(stage: OpeningStage): Promise<GroupOpeningStage> {
+        let
+            status: OpeningStatus | undefined,
+            stageBlock: number | undefined,
+            stageDate: Date | undefined;
+
+        if (stage.type === OpeningStageKeys.WaitingToBegin) {
```

Comment:
I think a better typesafe way of doing the comparison would be by using a helper methods or getters on the `OpeningStage` type to look something like:

```
## bool property
stage.isWaitingToBegin
```

or 

```
# I think the Enum type has an eq method
stage.eq(OpeningStage.WaitingToBeing)

# WaitingToBeing would be a static method on OpeningStage:
static WaitingToBeing() : OpeningStage  {
  return new OpeningStage('WaitingToBeing')
}
```

I've seen multiple instances where the comparison of the enum variant is done in this way.

Reply:
of course I'm simplifying here to express the idea, clearly the variant has encapsulated data as well which I'm disregarding

Reply:
if I remember correctly, we might even get some of these helper properties as part of the Enum type. I've seen some use of it in the polkadot-js/apps code base.